lib/storage/src/content_manager/consensus/persistent.rs

```diff
@@ -156,7 +156,9 @@ impl Persistent {
     /// `first_peer` - if this is a first peer in a new deployment (e.g. it does not bootstrap from anyone)
     /// It is `None` if distributed deployment is disabled
     fn init(path: PathBuf, first_peer: bool) -> Result<Self, StorageError> {
-        let this_peer_id = rand::random();
+        // Do not generate too big peer ID, to avoid problems with serialization
+        // (especially in json format)
+        let this_peer_id = rand::random::<PeerId>() % (1 << 53);
```

Comment:
We can do:

```diff
-let this_peer_id = rand::random::<PeerId>() % (1 << 53);
+let this_peer_id = rand::thread_rng().gen_range(0..1 << 53);
```

But I guess this is fine in this case.

Reply:
since you are not pushing, I am going to merge as is :smile: